---
layout: post
title: Pocillopora meandrina RNA-seq Bioinformatic Workflow
date: 2023-03-16
category: [ Code ]
tag: [ RNA-seq, Bioinformatics, Pocillopora meandrina ]
projects: E5 - Molecular Underpinnings
---
## Alignment comparison of the *Pocillopora meandrina* genome to the *Pocillopora verrucosa* genome for E5 molecular RNASeq data

### **Goal**

The following document contains the bioinformatic pipeline used for cleaning, aligning and assembling our raw RNA sequences. The goal is to compare the alignment statistics when using the Hawaiian *Pocillopora meandrina* genome compared to the Red Sea *Pocillopora verrucosa* genome. These commands were compiled into bash scripts to run on the URI HPC [Andromeda](https://its.uri.edu/research-computing/using-andromeda/) server.

### Project overview

**Bioinformatic tools used in analysis:**

- Quality check: [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/), [MultiQC](https://multiqc.info/)
- Quality trimming: [Fastp](https://github.com/OpenGene/fastp)
- Alignment to reference genome: [HISAT2](https://ccb.jhu.edu/software/hisat2/index.shtml)
- Preparation of alignment for assembly: [SAMtools](http://www.htslib.org/doc/samtools.html)
- Transcript assembly and quantification: [StringTie](https://ccb.jhu.edu/software/stringtie/)


### Check for required software on Andromeda
### Update software to latest version
- fastqc: FastQC/0.11.9-Java-11
- MultiQC: MultiQC/1.9-intel-2020a-Python-3.8.2
- fastp: fastp/0.19.7-foss-2018b
- HISAT2: HISAT2/2.2.1-foss-2019b
- SAMtools: SAMtools/1.9-foss-2018b
- StringTie: StringTie/2.2.1-GCC-11.2.0


### Prepare work space
- Upload raw reads and reference genome to server
- Assess that your files have all uploaded correctly
- Prepare your working directory
- Install all necessary programs

#### Upload raw reads and reference genome to server

This is done with the ```scp``` or "secure copy" linux command. SCP allows the secure transferring of files between a local host and a remote host or between two remote hosts using ssh authorization.

Secure Copy (scp) Options:
- -r - Recursively copy entire directories

```
scp -r xxxx <path_to_raw_reads> danielle_becker@bluewaves.uri.edu:<path_to_storage>

scp -r /Users/Danielle/Downloads/Pocillopora_meandrina_HIv1.assembly.fasta ssh danielle_becker@ssh3.hac.uri.edu:/data/putnamlab/REFS/Pmean/genome_assembly
```

# 1) Obtain Reference Genome and Create Folder Structure

I am using the Hawaiian *Pocillopora meandrina* genome to compare alignment statistics.

[Stephens, T. G., Lee, J., Jeong, Y., Yoon, H. S., Putnam, H. M., MajerovÃ¡, E., & Bhattacharya, D. (2022). High-quality genome assembles from key Hawaiian coral species. GigaScience, 11.](https://watermark.silverchair.com/giac098.pdf?token=AQECAHi208BE49Ooan9kkhW_Ercy7Dm3ZL_9Cf3qfKAc485ysgAAAxUwggMRBgkqhkiG9w0BBwagggMCMIIC_gIBADCCAvcGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMea0mgvgiBc31HuYiAgEQgIICyFGJOPhen3tUvhqtIxKl3OCxeM1Jdz-xMGvnPlWycY9E1F90MY4Oq4t1S_cVvGb1DU2AASTkgBB5CLG6PWFRwQ0x4DFAJnFwoQYqwqW8oWa6jf5bWx0T-alAdABe2vKaE8mCG6atHa-EscuuBYZUr4mjc185D2jfnyt0JCIEEflkpq97k6ZWu6ukObgcOdT4QzaeiCTPDQfJES45jYsX9--ZnnSyOfMEZhZ88Ck8fy5XCETQjY3CWl5Ae_0Hgm7VhX1liZxO-HRzyc03LNpql8J9en6utm0HuP66tL-wipOMGUikWr_JpWfUgmLMIa3fIJmrbzP1tcN7abKkTHy4VzZ-RsJfRIuT6Os9id86CWbDVZRba0HWOYBtZm0gENrVHVoPTGTWTklEXdjPvEOgPvkgVHRKLWY_RcNKvEZyzxlgj1hAfL4UmUdQYrIGbIpXH9Dw0yy5jP8z-LL5JVvUZ9OoqUWjnUfgtUQWHQ1oIGJvKeAQJ9gQ9bLE0E8eqnSFnX9F1xwEcRk4bPYWPBkN7Q6cxpBhCRAQM5pyvg9S8Hyx7mzzHylYYEmnJUjhB-h13qNpv5GGYgQ8SQDqmBkGFx7nwRZ2aEvbj8fMAf-pky5TrV3xJl7vbdMtOeeqqpW4Af0L2VIa1lLlwGEFtYXs3WLG8BZHIkpSFuhbja1lK5G2fjeiwfIA4ZfiVBpiCaNPjQgkHfMA-0-yt92zhgn1rV9a1Wj5Ku3gNlmtbVA3PzLlGzzQKFXLak54bdGVC24rx0tSDJYlRPAF4l5wNiJS2tlbm7DEvueKNZlBWZICkDrfik-93NHt3sHZTao2fG1pe08skWpXoOZInsBBGGc1j13lx8ABe7f2OraI4CVByebNfidaqzcEPPql8_vUNpjWNe5bwF2On0ziX3ESqN_ADJROK00dkXsmocA6Ia1phOLu8Bfg8Sa5_Dc)

Location on Andromeda, the HPC server for URI:
```
cd /data/putnamlab/REFS/

mkdir Pmean
```

All of the genome files (i.e., genome scaffolds, CDS, proteins, GFF, and functional annotations) were downloaded from the [Rutgers online database](http://cyanophora.rutgers.edu/Pocillopora_meandrina/).

I put them on the Putnam Lab HPC Andromeda server for future reference:

```
pwd /data/putnamlab/REFS/Pmean/
```

You can also directly download the genome files from the [Rutgers online database](http://cyanophora.rutgers.edu/Pocillopora_meandrina/):

```
cd /data/putnamlab/REFS/Pmean/data/refs/

wget http://cyanophora.rutgers.edu/Pocillopora_meandrina/Pocillopora_meandrina_HIv1.assembly.fasta.gz

wget http://cyanophora.rutgers.edu/Pocillopora_meandrina/Pocillopora_meandrina_HIv1.genes.gff3.gz
```

Unzip gff and genome file

```
gunzip Pocillopora_meandrina_HIv1.assembly.fasta.gz
gunzip Pocillopora_meandrina_HIv1.genes.gff3.gz

```

### Make folder structure in personal pathway on Andromeda
```
pwd /data/putnamlab/dbecks/Becker_E5/

mkdir Pmean_RNASeq_compare

cd Pmean_RNASeq_compare

mkdir data/
mkdir scripts/

```

### Download files into data folder
#### Path where we stored the RAW fastq.gz files

```
/data/putnamlab/KITT/hputnam/20201209_Becker_RNASeq_combo/combo
```

#### Use code to copy files from one directory to the other

```
cp -R /data/putnamlab/KITT/hputnam/20201209_Becker_RNASeq_combo/combo /data/putnamlab/dbecks/Becker_E5/Pmean_RNASeq_compare/data/raw
```

# 2) Check file integrity

a) Count all files to make sure all downloaded

```
cd data/raw

ls -1 | wc -l
```

b) Verify data transfer integrity with md5sum

```
md5sum *.fastq > URIcheckmd5.md5
md5sum -c URIcheckmd5.md5
```
Should output 'OK' next to each file name


d) Count number of reads per file using the code after @ in fastq.gz files (e.g.,@GWNJ).

```
zgrep -c "@GWNJ" *.gz > raw_seq_counts

```

```
C17_R1_001.fastq.gz:25158606
C17_R2_001.fastq.gz:25158606
C18_R1_001.fastq.gz:22733345
C18_R2_001.fastq.gz:22733345
C19_R1_001.fastq.gz:24846067
C19_R2_001.fastq.gz:24846067
C20_R1_001.fastq.gz:24030431
C20_R2_001.fastq.gz:24030431
C21_R1_001.fastq.gz:16484060
C21_R2_001.fastq.gz:16484060
C22_R1_001.fastq.gz:22990550
C22_R2_001.fastq.gz:22990550
C23_R1_001.fastq.gz:20905338
C23_R2_001.fastq.gz:20905338
C24_R1_001.fastq.gz:22578178
C24_R2_001.fastq.gz:22578178
C25_R1_001.fastq.gz:29417106
C25_R2_001.fastq.gz:29417106
C26_R1_001.fastq.gz:23267238
C26_R2_001.fastq.gz:23267238
C27_R1_001.fastq.gz:24990687
C27_R2_001.fastq.gz:24990687
C28_R1_001.fastq.gz:23396439
C28_R2_001.fastq.gz:23396439
C29_R1_001.fastq.gz:17900262
C29_R2_001.fastq.gz:17900262
C30_R1_001.fastq.gz:26361873
C30_R2_001.fastq.gz:26361873
C31_R1_001.fastq.gz:24642131
C31_R2_001.fastq.gz:24642131
C32_R1_001.fastq.gz:27076800
C32_R2_001.fastq.gz:27076800
E10_R1_001.fastq.gz:29131006
E10_R2_001.fastq.gz:29131006
E11_R1_001.fastq.gz:18568153
E11_R2_001.fastq.gz:18568153
E12_R1_001.fastq.gz:27805087
E12_R2_001.fastq.gz:27805087
E13_R1_001.fastq.gz:24455094
E13_R2_001.fastq.gz:24455094
E14_R1_001.fastq.gz:22630044
E14_R2_001.fastq.gz:22630044
E15_R1_001.fastq.gz:23796710
E15_R2_001.fastq.gz:23796710
E16_R1_001.fastq.gz:29523059
E16_R2_001.fastq.gz:29523059
E1_R1_001.fastq.gz:25368402
E1_R2_001.fastq.gz:25368402
E2_R1_001.fastq.gz:23770610
E2_R2_001.fastq.gz:23770610
E3_R1_001.fastq.gz:25641209
E3_R2_001.fastq.gz:25641209
E4_R1_001.fastq.gz:21751368
E4_R2_001.fastq.gz:21751368
E5_R1_001.fastq.gz:16381619
E5_R2_001.fastq.gz:16381619
E6_R1_001.fastq.gz:24937261
E6_R2_001.fastq.gz:24937261
E7_R1_001.fastq.gz:24020166
E7_R2_001.fastq.gz:24020166
E8_R1_001.fastq.gz:23675842
E8_R2_001.fastq.gz:23675842
E9_R1_001.fastq.gz:25068848
E9_R2_001.fastq.gz:25068848
=======

/data/putnamlab/KITT/hputnam/20201209_Becker_RNASeq_combo/combo/md5sum_list.txt
```

# 3) Run FastQC

a) Make folders for raw FastQC results and scripts

```
cd Francois_data/Hawaii

mkdir fastqc_results/raw

```

b) Write script for checking quality with FastQC and submit as job on Andromeda

```
nano /data/putnamlab/dbecks/Becker_E5/Pmean_RNASeq_compare/scripts/fastqc_raw.sh
```

```  
#!/bin/bash
#SBATCH -t 24:00:00
#SBATCH --nodes=1 --ntasks-per-node=1
#SBATCH --export=NONE
#SBATCH --mem=100GB
#SBATCH --mail-type=BEGIN,END,FAIL #email you when job starts, stops and/or fails
#SBATCH --mail-user=danielle_becker@uri.edu #your email to send notifications
#SBATCH --account=putnamlab
#SBATCH -D /data/putnamlab/dbecks/Becker_E5/Pmean_RNASeq_compare/data/raw
#SBATCH --error="script_error" #if your job fails, the error report will be put in this file
#SBATCH --output="output_script" #once your job is completed, any final job report comments will be put in this file

module load FastQC/0.11.9-Java-11

for file in /data/putnamlab/dbecks/Becker_E5/Pmean_RNASeq_compare/data/raw/*.gz
do
fastqc $file --outdir /data/putnamlab/dbecks/Becker_E5/Pmean_RNASeq_compare/data/raw/qc
done
```

```
sbatch /data/putnamlab/dbecks/Becker_E5/Pmean_RNASeq_compare/scripts/fastqc_raw.sh
```
Submitted batch job 1816766 on 20210104



c) Make sure all files were processed

```
ls -1 | wc -l
#64
```

## Combined QC output into 1 file with MultiQC

```
module load MultiQC/1.9-intel-2020a-Python-3.8.2 /data/putnamlab/dbecks/Becker_E5/Pmean_RNASeq_compare/data/raw/qc

```

c) Copy MultiQC files to local computer

```
scp -r danielle_becker@ssh3.hac.uri.edu:/data/putnamlab/dbecks/Becker_E5/Pmean_RNASeq_compare/data//trimmed/trimmed_qc/multiqc_report.html /Users/Danielle/Desktop/Putnam_Lab/Becker_E5/Bioinformatics/Output/RNASeq/trimmed.qc.multiqc

```


# 4) Trim and clean reads

a) Make trimmed reads folder in all other results folders

```

mkdir data/trimmed
cd trimmed

```

c) Write script for Trimming and run on Andromeda

#Run fastp on files
#Trims 20bp from 5' end of all reads
#Trims poly G, if present

```
nano /data/putnamlab/dbecks/Becker_E5/Pmean_RNASeq_compare/scripts/trim.sh
```

```
#!/bin/bash
#SBATCH -t 24:00:00
#SBATCH --nodes=1 --ntasks-per-node=1
#SBATCH --export=NONE
#SBATCH --mem=100GB
#SBATCH --mail-type=BEGIN,END,FAIL #email you when job starts, stops and/or fails
#SBATCH --mail-user=danielle_becker@uri.edu #your email to send notifications
#SBATCH --account=putnamlab
#SBATCH -D /data/putnamlab/dbecks/Becker_E5/Pmean_RNASeq_compare/data/raw
#SBATCH --error="script_error" #if your job fails, the error report will be put in this file
#SBATCH --output="output_script" #once your job is completed, any final job report comments will be put in this file

module load fastp/0.19.7-foss-2018b

array1=($(ls *R1*.fastq.gz)) #Make an array of sequences to trim
for i in ${array1[@]}; do
fastp --in1 ${i} --in2 $(echo ${i}|sed s/_R1/_R2/) --detect_adapter_for_pe --trim_poly_g --trim_front1 20 --trim_front2 20 --out1 ../trimmed/${i} --out2 ../trimmed/$(echo ${i}|sed s/_R1/_R2/)  
done

```
```
sbatch /data/putnamlab/dbecks/Becker_E5/Pmean_RNASeq_compare/scripts/trim.sh
Submitted batch job 235101 on 20220223 - started at 15:30 pm, ended at 01:30 am - 10 hours
```


# 5) Check quality of trimmed files

a) Check number of files in /trimmed directory

```
ls -1 | wc -l
#64
```


b) Check number of reads in /trimmed directory

```
zgrep -c "@GWNJ" *.gz > trimmed_seq_counts

C17_R1.fastp-trim.20230215.fq.gz:24075726
C17_R2.fastp-trim.20230215.fq.gz:24075726
C18_R1.fastp-trim.20230215.fq.gz:22164069
C18_R2.fastp-trim.20230215.fq.gz:22164069
C19_R1.fastp-trim.20230215.fq.gz:24108648
C19_R2.fastp-trim.20230215.fq.gz:24108648
C20_R1.fastp-trim.20230215.fq.gz:23308448
C20_R2.fastp-trim.20230215.fq.gz:23308448
C21_R1.fastp-trim.20230215.fq.gz:15900108
C21_R2.fastp-trim.20230215.fq.gz:15900108
C22_R1.fastp-trim.20230215.fq.gz:22276164
C22_R2.fastp-trim.20230215.fq.gz:22276164
C23_R1.fastp-trim.20230215.fq.gz:20264849
C23_R2.fastp-trim.20230215.fq.gz:20264849
C24_R1.fastp-trim.20230215.fq.gz:21739688
C24_R2.fastp-trim.20230215.fq.gz:21739688
C25_R1.fastp-trim.20230215.fq.gz:28400490
C25_R2.fastp-trim.20230215.fq.gz:28400490
C26_R1.fastp-trim.20230215.fq.gz:22356889
C26_R2.fastp-trim.20230215.fq.gz:22356889
C27_R1.fastp-trim.20230215.fq.gz:23741291
C27_R2.fastp-trim.20230215.fq.gz:23741291
C28_R1.fastp-trim.20230215.fq.gz:22662492
C28_R2.fastp-trim.20230215.fq.gz:22662492
C29_R1.fastp-trim.20230215.fq.gz:17264107
C29_R2.fastp-trim.20230215.fq.gz:17264107
C30_R1.fastp-trim.20230215.fq.gz:25439860
C30_R2.fastp-trim.20230215.fq.gz:25439860
C31_R1.fastp-trim.20230215.fq.gz:23852854
C31_R2.fastp-trim.20230215.fq.gz:23852854
C32_R1.fastp-trim.20230215.fq.gz:25904515
C32_R2.fastp-trim.20230215.fq.gz:25904515
E10_R1.fastp-trim.20230215.fq.gz:28094072
E10_R2.fastp-trim.20230215.fq.gz:28094072
E11_R1.fastp-trim.20230215.fq.gz:17954806
E11_R2.fastp-trim.20230215.fq.gz:17954806
E12_R1.fastp-trim.20230215.fq.gz:26996226
E12_R2.fastp-trim.20230215.fq.gz:26996226
E13_R1.fastp-trim.20230215.fq.gz:23732695
E13_R2.fastp-trim.20230215.fq.gz:23732695
E14_R1.fastp-trim.20230215.fq.gz:21831061
E14_R2.fastp-trim.20230215.fq.gz:21831061
E15_R1.fastp-trim.20230215.fq.gz:22614254
E15_R2.fastp-trim.20230215.fq.gz:22614254
E16_R1.fastp-trim.20230215.fq.gz:28776838
E16_R2.fastp-trim.20230215.fq.gz:28776838
E1_R1.fastp-trim.20230215.fq.gz:24465103
E1_R2.fastp-trim.20230215.fq.gz:24465103
E2_R1.fastp-trim.20230215.fq.gz:23120087
E2_R2.fastp-trim.20230215.fq.gz:23120087
E3_R1.fastp-trim.20230215.fq.gz:25025833
E3_R2.fastp-trim.20230215.fq.gz:25025833
E4_R1.fastp-trim.20230215.fq.gz:21164262
E4_R2.fastp-trim.20230215.fq.gz:21164262
E5_R1.fastp-trim.20230215.fq.gz:15836446
E5_R2.fastp-trim.20230215.fq.gz:15836446
E6_R1.fastp-trim.20230215.fq.gz:24285839
E6_R2.fastp-trim.20230215.fq.gz:24285839
E7_R1.fastp-trim.20230215.fq.gz:23378626
E7_R2.fastp-trim.20230215.fq.gz:23378626
E8_R1.fastp-trim.20230215.fq.gz:22972241
E8_R2.fastp-trim.20230215.fq.gz:22972241
E9_R1.fastp-trim.20230215.fq.gz:24270035
E9_R2.fastp-trim.20230215.fq.gz:24270035


scp -r danielle_becker@ssh3.hac.uri.edu:/data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/trimmed/trimmed_seq_counts /Users/Danielle/Desktop/Putnam_Lab/Becker_E5/Bioinformatics/Output/RNASeq/

```



c) Run FastQC on trimmed data
```
mkdir trimmed_qc

```
```
nano /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/fastqc_trimmed.sh
```

```  
#!/bin/bash
#SBATCH -t 24:00:00
#SBATCH --nodes=1 --ntasks-per-node=1
#SBATCH --export=NONE
#SBATCH --mem=100GB
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=danielle_becker@uri.edu
#SBATCH --account=putnamlab
#SBATCH -D /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/trimmed/trimmed_qc
#SBATCH --error="script_error"
#SBATCH --output="output_script"

module load FastQC/0.11.8-Java-1.8

for file in /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/trimmed/*.gz
do
fastqc $file --outdir /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/trimmed/trimmed_qc
done
```

```
sbatch /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/fastqc_trimmed.sh
Submitted batch job 1834516
```


d) Run MultiQC on trimmed data
```
module load MultiQC/1.9-intel-2020a-Python-3.8.2
multiqc /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/trimmed/trimmed_qc
```
```
scp -r danielle_becker@ssh3.hac.uri.edu:/data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/trimmed/trimmed_qc/*.html /Users/Danielle/Desktop/Putnam_Lab/Becker_E5/Bioinformatics/Output/RNASeq/trimmed.qc.multiqc

```

# 6) Align reads

a) Generate genome build

### HiSat2 Align reads to reference genome
[HiSat2](https://daehwankimlab.github.io/hisat2/main/)
[HiSat2 Github](https://github.com/DaehwanKimLab/hisat2)

```
nano /data/putnamlab/dbecks/Becker_E5/Pmean_RNASeq_compare/scripts/Hisat2_genome_build.sh
```

```
#!/bin/bash
#SBATCH -t 24:00:00
#SBATCH --nodes=1 --ntasks-per-node=1
#SBATCH --export=NONE
#SBATCH --mem=100GB
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=danielle_becker@uri.edu
#SBATCH --account=putnamlab
#SBATCH -D /data/putnamlab/dbecks/Becker_E5/Pmean_RNASeq_compare/data/refs
#SBATCH --error="script_error"
#SBATCH --output="output_script"

# load modules needed
module load HISAT2/2.2.1-foss-2019b
module load SAMtools/1.9-foss-2018b

#unzip reference genome
#gunzip Montipora_capitata_HIv3.assembly.fasta.gz

hisat2-build -f /data/putnamlab/dbecks/Becker_E5/Pmean_RNASeq_compare/data/refs/Pmean/ ./Pmean_ref
echo "Referece genome indexed. Starting alingment" $(date)


# This script exports alignments as bam files
# sorts the bam file because Stringtie takes a sorted file for input (--dta)
# removes the sam file because it is no longer needed
array1=($(ls *R1*.fq.gz)) # call the clean sequences - make an array to align
for i in ${array1[@]}; do
        sample_name=`echo $i| awk -F [.] '{print $2}'`
	hisat2 -p 8 --dta -x Mcapitata_ref_v3 -U ${i} -S ${sample_name}.sam
        samtools sort -@ 8 -o ${sample_name}.bam ${sample_name}.sam
    		echo "${i} bam-ified!"
        rm ${sample_name}.sam
done

array1=($(ls *R1*.fq.gz))
for i in ${array1[@]}; do
hisat2 -p 48 --rna-strandness RF --dta -q -x /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/refs/Pver_ref -1 /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/trimmed/${i} \
-2 /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/trimmed/$(echo ${i}|sed s/_R1/_R2/) -S /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/${i}.sam
done

```
```
sbatch /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/Hisat2_genome_build.sh
Submitted batch job 1834918
```

b) Align reads to genome

```
mkdir mapped
```
```
nano /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/Hisat2_align2.sh
```

```
#!/bin/bash
#SBATCH -t 72:00:00
#SBATCH --nodes=1 --ntasks-per-node=5
#SBATCH --export=NONE
#SBATCH --mem=500GB
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=danielle_becker@uri.edu
#SBATCH --account=putnamlab
#SBATCH -D /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/trimmed
#SBATCH -p putnamlab
#SBATCH --cpus-per-task=3
#SBATCH --error="script_error"
#SBATCH --output="output_script"


module load HISAT2/2.1.0-foss-2018b

#Aligning paired end reads
#Has the R1 in array1 because the sed in the for loop changes it to an R2. SAM files are of both forward and reverse reads

array1=($(ls *R1*.fq.gz))
for i in ${array1[@]}; do
hisat2 -p 48 --rna-strandness RF --dta -q -x /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/refs/Pver_ref -1 /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/trimmed/${i} \
-2 /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/trimmed/$(echo ${i}|sed s/_R1/_R2/) -S /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/${i}.sam
done
```

```
sbatch /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/Hisat2_align2.sh
#Submitted batch job 235530 20230223 at 17:14, ended at 21:18, 3 hours 44 minutes

#Download alignment statistics information from mapping, will be the output from your script above, even though it is a mapped output, you are in the trimmed folder here

scp -r danielle_becker@ssh3.hac.uri.edu:/data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/trimmed/script_error /Users/Danielle/Desktop/Putnam_Lab/Becker_E5/Bioinformatics/Output/RNASeq/

```

## Sort and convert sam to bam

```
nano /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/SAMtoBAM.sh

#There will be lots of .tmp file versions in your folder, this is normal while this script runs and they should delete at the end to make one sorted.bam file
```
```
#!/bin/bash
#SBATCH -t 72:00:00
#SBATCH --nodes=1 --ntasks-per-node=8
#SBATCH --export=NONE
#SBATCH --mem=500GB
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=danielle_becker@uri.edu
#SBATCH --account=putnamlab
#SBATCH -D /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped
#SBATCH --cpus-per-task=3
#SBATCH --error="script_error"
#SBATCH --output="output_script"

module load SAMtools/1.9-foss-2018b

array1=($(ls *.sam))  
for i in ${array1[@]}; do
samtools sort -o /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/${i}.sorted.bam /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/${i}  
done
```

```
sbatch /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/SAMtoBAM.sh
#Submitted batch job 235853 on 20230224 at 09:02, finished at 06:51, was 18 hours total
```

### Remove Sam files to save space
```
rm /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/*.sam

```


### Check number of mapped reads
```
Explanation for SAMtools functions for checking the mapped reads in a paired-end dataset found here: https://www.biostars.org/p/138116/

nano /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/mapped_read_counts.sh

#!/bin/bash
#SBATCH -t 72:00:00
#SBATCH --nodes=1 --ntasks-per-node=8
#SBATCH --export=NONE
#SBATCH --mem=500GB
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=danielle_becker@uri.edu
#SBATCH --account=putnamlab
#SBATCH -D /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped
#SBATCH --cpus-per-task=3

module load SAMtools/1.9-foss-2018b

array1=($(ls *.bam))  
for i in ${array1[@]}; do
samtools view -F 0x4 /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/${i} | cut -f 1 | sort | uniq | wc -l > mapped_read_counts
done

sbatch /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/mapped_read_counts.sh

#Submitted batch job 236602, took 7 hours
#19421210 counts

```

# 7) Perform gene counts with stringTie

### Needed to modify Pverr_genome_assembly file, information in this [GitHub issue](https://github.com/Putnam-Lab/Lab_Management/issues/11)

```
##copy modified genome assembly file to Andromeda, enter this command into local computer shell

scp -r /Users/Danielle/Downloads/Pver_genome_assembly_v1.0_modified.gff3 danielle_becker@Andromeda.uri.edu:/data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/refs/Pverr/
```

```
mkdir counts
cd counts

```


b) Assemble and estimate reads

```
nano /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/StringTie_Assemble.sh
```

```
#!/bin/bash
#SBATCH -t 72:00:00
#SBATCH --nodes=1 --ntasks-per-node=5
#SBATCH --export=NONE
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=danielle_becker@uri.edu
#SBATCH --account=putnamlab
#SBATCH -D /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped
#SBATCH --cpus-per-task=3

module load StringTie/2.2.1-GCC-11.2.0

array1=($(ls *.bam))
for i in ${array1[@]}; do
stringtie -p 48 --rf -e -G /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/refs/Pverr/Pver_genome_assembly_v1.0_modified.gff3 -o /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/counts/${i}.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/${i}
done
```

```
sbatch /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/StringTie_Assemble.sh
Submitted batch job 237016, 1 hour 20 minutes
```

c) Merge stringTie gtf results

#in this step we are making a file with all the gtf names and stringtie will merge them all together for a master list for your specific genes

```
ls *gtf > mergelist.txt
cat mergelist.txt

module load StringTie/2.2.1-GCC-11.2.0

stringtie --merge -p 8 -G /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/refs/Pverr/Pver_genome_assembly_v1.0_modified.gff3 -o stringtie_merged.gtf mergelist.txt

```

d) Assess assembly quality

```

nano /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/gffcompare.sh

#!/bin/bash
#SBATCH -t 72:00:00
#SBATCH --nodes=1 --ntasks-per-node=8
#SBATCH --export=NONE
#SBATCH --mem=500GB
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=danielle_becker@uri.edu
#SBATCH --account=putnamlab
#SBATCH -D /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/counts
#SBATCH --cpus-per-task=3

module load GffCompare/0.12.6-GCC-11.2.0

gffcompare -r /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/refs/Pverr/Pver_genome_assembly_v1.0_modified.gff3 -o merged stringtie_merged.gtf

sbatch /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/gffcompare.sh

#Submitted batch job 237924, took one minute

```

e) Re-estimate assembly

```
nano /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/re_estimate.assembly.sh
```
```
#!/bin/bash
#SBATCH -t 72:00:00
#SBATCH --nodes=1 --ntasks-per-node=5
#SBATCH --export=NONE
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=danielle_becker@uri.edu
#SBATCH --account=putnamlab
#SBATCH -D /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped
#SBATCH --cpus-per-task=3

module load StringTie/2.2.1-GCC-11.2.0

array1=($(ls *.bam))
for i in ${array1[@]}; do
stringtie -e -G /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/refs/Pverr/Pver_genome_assembly_v1.0_modified.gff3 -o ${i}.merge.gtf ${i}
echo "${i}"
done

```
```
sbatch /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/re_estimate.assembly.sh
Submitted batch job 237925, 1 hour 24 minutes
```

```
# move merged GTF files to their own folder
mkdir GTF_merge

mv *merge.gtf GTF_merge

```

f) Create gene matrix


```
#making a sample txt file with all gtf file names

F=/data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/

array2=($(ls *merge.gtf))
for i in ${array2[@]}
do
echo "${i} $F${i}" >> sample_list.txt
done

```
```
#sample_list.txt document

C17_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C17_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C18_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C18_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C19_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C19_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C20_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C20_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C21_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C21_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C22_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C22_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C23_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C23_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C24_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C24_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C25_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C25_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C26_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C26_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C27_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C27_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C28_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C28_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C29_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C29_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C30_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C30_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C31_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C31_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
C32_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/C32_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E10_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E10_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E11_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E11_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E12_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E12_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E13_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E13_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E14_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E14_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E15_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E15_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E16_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E16_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E1_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E1_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E2_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E2_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E3_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E3_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E4_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E4_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E5_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E5_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E6_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E6_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E7_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E7_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E8_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E8_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf
E9_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/E9_R1.fastp-trim.20230215.fq.gz.sam.sorted.bam.merge.gtf

```
```
#create gene matrix
```

```
nano /data/putnamlab/hputnam/Becker_E5/RNASeq_Becker_E5/scripts/GTFtoCounts.sh
```

```
#!/bin/bash
#SBATCH -t 72:00:00
#SBATCH --nodes=1 --ntasks-per-node=5
#SBATCH --export=NONE
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=danielle_becker@uri.edu
#SBATCH --account=putnamlab
#SBATCH -D /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge
#SBATCH --cpus-per-task=3


module load StringTie/2.2.1-GCC-11.2.0
module load Python/2.7.18-GCCcore-9.3.0

python prepDE.py -g Poc_gene_count_matrix.csv -i sample_list.txt

```

```
sbatch /data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/scripts/GTFtoCounts.sh
Submitted batch job 237934
```


g) Secure-copy gene counts onto local computer, make sure to open a seperate command shell outside of Andromeda on your own terminal

```
#copy gene count matrix

scp danielle_becker@ssh3.hac.uri.edu:/data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/Poc_gene_count_matrix.csv /Users/Danielle/Desktop/Putnam_Lab/Becker_E5/RAnalysis/Data/RNA-seq/Host


#copy transcript count matrix
scp danielle_becker@ssh3.hac.uri.edu:/data/putnamlab/dbecks/Becker_E5/Becker_RNASeq/data/mapped/GTF_merge/transcript_count_matrix.csv /Users/Danielle/Desktop/Putnam_Lab/Becker_E5/RAnalysis/Data/RNA-seq/Host

```
