---
layout: post
title: August 2024 Goals
date: '2024-08-01'
categories: Monthly_Goals
tags: Goals
---

August 2024 goals and daily notebook entries. 

## Goals  

### Writing and manuscripts 
              
1. Submit molecular underpinnings (Ch.1) to BioRXiv by end of month
2. A.pul gametogenesis multivariate analysis
3. Marine heatwaves manuscript transcriptome and gene expression

### Sample processing

1. Library prep for heatwave WGBS samples (12)?

### Data analysis

1. Work on DEG analysis for heatwave 
2. Variance partitioning/redundancy analyses and condition indices for Ch2
3. Ch3 gene expression

## Daily log 

### August 1
- Updated methods and results for transcriptome, ran cd-hit-est for transcriptome
- Updated results and methods for ch2
- Updated DEG and DMG results and discussion sections for Ch1 from topGO

### August  2
- Redid topGO for DEGs with Hollie suggestions
- Responded and addressed Hollies edits and comments on methods and results sections for Ch1
- New approach for topGO

### August 3
- Redid topGO for DEGs with Hollie suggestions
- Responded and addressed Hollies edits and comments on methods and results sections for Ch1
- New approach for topGO for ch1, updated results sections and made tables for top DEGs

### August 5
- Ch1 genes of interest list compilation
- Confirmed defense date and time with committee members
- Ch3 transcriptome mapping workflow, functional annotation workflow
- Updated hypothesis and edits for ch2 and ch3
- New approach for topGO for ch1, updated results sections and made tables for top DEGs

### August 6
- Ch1 genes of interest list compilation and literature review 
- Ch3 transcriptome mapping workflow to updated A.pul genome, functional annotation workflow and literature review for [Trinotate](https://github.com/Trinotate/Trinotate/wiki/Software-installation-and-data-required) approach
- Updated hypothesis and edits for ch2 and ch3

### August 7
- Aquarium duties, adjusted alkalinity outputs
- Ch1 genes of interest list compilation and literature review 
- Slide scanning at Brown for new POC samples
- Kelp timeseries manuscript and MERMAID data entry
- Aligned reads to de novo transcriptome for ch3, running RSEM for all samples

### August 8
- Aquarium duties
- Ch1 genes of interest list compilation and literature review 
- Kelp timeseries manuscript and MERMAID data entry
- Continued workflow for running RSEM for samples, CH3
- Updated correlation figures for DEG and DMG for CH1
- Rewrote results sections for Ch1 for DEG topGO
- Rerunning BS-SNPer with only 29 samples instead of 32

### August 9
- Ch1 genes of interest list compilation and literature review 
- Literature review for Ch3 transcriptome functional annotation, began workflow on Andromeda for [trinotate](https://github.com/Trinotate/Trinotate/wiki/Software-installation-and-data-required)
- Reworked ch2 hypotheses with literature based approaches for more targeted approach than overarching correlations across all timepoints
- Rerunning BS-SNPer with only 29 samples instead of 32

### August 10
- Comments and review of CEABiGR Manuscript for requested re-submission of revisions

### August 11
- Comments and review of CEABiGR Manuscript for requested re-submission of revisions

### August 12
- Ch1 genes of interest list compilation completion
- Literature review for Ch3 transcriptome functional annotation, continued working on workflow on Andromeda for [trinotate](https://github.com/Trinotate/Trinotate/wiki/Software-installation-and-data-required)
- Reworked ch2 hypotheses with literature based approaches for more targeted approach than overarching correlations across all timepoints
- Rerunning BS-SNPer with only 29 samples instead of 32

### August 13
- Ch1 correlation plots, DEG, DMG comparisons
- topGO for DMGs and genes of interest compilation
- Literature review for Ch3 transcriptome functional annotation, continued working on workflow on Andromeda for [trinotate](https://github.com/Trinotate/Trinotate/wiki/Software-installation-and-data-required)
- Reworked ch2 hypotheses with literature based approaches for more targeted approach than overarching correlations across all timepoints

### August 14
- COVID rest day

### August 15
- Ch1 correlation plots, DEG, DMG comparisons
- topGO for DMGs and genes of interest compilation
- Literature review for Ch3 transcriptome functional annotation, continued working on workflow on Andromeda for [trinotate](https://github.com/Trinotate/Trinotate/wiki/Software-installation-and-data-required)
- Reworked ch2 hypotheses with literature based approaches for more targeted approach than overarching correlations across all timepoints
- Ch2 updated introduction final paragraph

### August 16
- Ch1 updated results and discussion paragraphs with new data and filtering step
- Ch2 wrote size and relative abundance results sections, updated introduction paragraphs
- Ch2 updated size and relative abundance figures, made stacked timeseries plot with data
- Ch3 noticed alignment rates through RSEM and bowtie2 lower than Hisat2, looking into literature and code

### August 17
Continued working on:
- Ch1 updated results and discussion paragraphs with new data and filtering step
- Ch2 wrote size and relative abundance results sections, updated introduction paragraphs
- NSF annual report
- Ch3 HISAT2 and stringtie workflow

### August 18
Continued working on:
- Ch1 updated results and discussion paragraphs with new data and filtering step
- NSF annual report
- Ch3 HISAT2 and stringtie workflow

### August 19
- Ch1 updated results and discussion paragraphs with new data and filtering step
- Ch3 HISAT2 and stringtie workflow
- NSF annual report

### August 20
Continued working on:
- Ch1 updated results and discussion paragraphs with new data and filtering step
- NSF annual report
- Ch3 stringtie and assembly

### August 21
Continued working on:
- Ch1 updated results and discussion paragraphs with new data and filtering step
- NSF annual report
- Ch3 stringtie workflow
- Ch2 updated size figures for NSF report





















