---
layout: post
title: Acropora pulchra Functional Annotation Workflow
date: 2024-08-02
category: [ Code ]
tag: [ RNA-seq, Bioinformatics, Acropora pulchra de novo transcriptome ]
projects: Heatwave - Gametogenesis ; E5 - Molecular Underpinnings
---

## Creating a functional annotation file for *Acropora pulchra* de novo transcriptome.

### **Goal**

The following document contains the bioinformatic pipeline used for the functional annotation of a *de novo* transcriptome for *A. pulchra*. Pipeline adapted from the [Trinotate wiki](https://github.com/Trinotate/Trinotate/wiki/Software-installation-and-data-required). Trinotate is a comprehensive annotation suite designed for automatic functional annotation of transcriptomes, particularly de novo assembled transcriptomes, from model or non-model organisms. Trinotate makes use of a number of different well referenced methods for functional annotation including homology search to known sequence data (BLAST+/SwissProt), protein domain identification (HMMER/PFAM), protein signal peptide and transmembrane domain prediction (signalP/tmHMM), and leveraging various annotation databases (eggNOG/GO/Kegg databases). All functional annotation data derived from the analysis of transcripts is integrated into a SQLite database which allows fast efficient searching for terms with specific qualities related to a desired scientific hypothesis or a means to create a whole annotation report for a transcriptome.

All metadata and information for these projects can be found in this [repository](https://github.com/daniellembecker/A.pul_Heatwave/tree/master) and in these notebook posts of the [*de novo* transcriptome assembly](https://github.com/daniellembecker/DanielleBecker_Lab_Notebook/blob/master/_posts/2023-08-31-Acropora-pulchra-denovo-transcriptome.md) and the [extraction protocol](https://github.com/daniellembecker/DanielleBecker_Lab_Notebook/blob/master/_posts/2023-04-25-Acropora-pulchra-transcriptome-extraction-concentration.md) for the samples used in the assembly. These commands were compiled into bash scripts to run on the URI HPC [Andromeda](https://its.uri.edu/research-computing/using-andromeda/) server.


# Step 1: Obtain *de novo* transcriptome

I am using the *Acropora pulchra* *de novo* transcriptome assembled in [November 2023](https://github.com/daniellembecker/DanielleBecker_Lab_Notebook/blob/master/_posts/2023-08-31-Acropora-pulchra-denovo-transcriptome.md), following methods outlined for [transcript sequence reconstruction from RNA-Seq: reference generation and analysis with Trinity](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3875132/).

Location on Andromeda, the HPC server for URI:
```
cd /data/putnamlab/dbecks/DeNovo_transcriptome/2023_A.pul/output/final_assembly/CD-HIT/stranded_assembly_cdhit.fasta

```

## Make folder structure for functional annotation pipeline

```
cd /data/putnamlab/dbecks/DeNovo_transcriptome/2023_A.pul/

mkdir functional_annotation
cd functional_annotation

mkdir Trinotate
cd Trinotate

mkdir data
mkdir output
mkdir scripts

```

# Step 2: Access previously created open reading frame prediction and protein sequence prediction with Transdecoder and Transpredictor

Already created .gff3, .pep, and .cds files for the reference transcriptome in step 8 of my [A.pulchra RNASeq Workflow](https://github.com/daniellembecker/DanielleBecker_Lab_Notebook/blob/master/_posts/2023-01-01-A.pulchra-RNASeq-Workflow.md).

Steps followed:

#### 1) Create .gff3 files for predicted gene structures, .pep for predicted proteins, and .cds predicted coding sequences for your de novo transcriptome

#To include putative gene information in your Trinity analysis, you can use TransDecoder.LongOrfs and TransDecoder.Predict
#First step identifies likely coding regions (long open reading frames or ORFs) in your Trinity transcripts and creates a file named Trinity.fasta.transdecoder_dir/longest_orfs.pep, which contains the predicted protein sequences.

#Second step predicts likely coding regions and identifies potential coding regions using the output from the LongOrfs step and generates several output files in the Trinity.fasta.transdecoder_dir/ directory, including Trinity.fasta.transdecoder.cds (predicted coding sequences) and Trinity.fasta.transdecoder.gff3 (predicted gene structures in GFF3 format).

Directory for TransDecoder and Transpredictor output files:

```
cd /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/data/trinity/

```

# Step 3: Set up Trinotate environmental variables in Andromeda HPC

Export the location of our Trinotate installation so we can run the software from any directory:

```
export TRINOTATE_HOME=/opt/software/Trinotate/4.0.2-foss-2022a
export PATH=$TRINOTATE_HOME:$PATH

```

# Step 4: Load the sequence databases required for TRINOTATE_HOME

Trinotate relies heavily on SwissProt and Pfam, and custom protein files are generated as described below to be specifically used with Trinotate. You can obtain the protein database files by running this Trinotate build process. This step will download several data resources including the most current versions of swissprot, pfam, and other companion resources, create and populate a Trinotate boilerplate sqlite database (Trinotate.sqlite), and yield uniprot_sprot.pep file to be used with BLAST, and the Pfam-A.hmm.gz file to be used for Pfam searches. Run the build process like so:

Create script and run:

```
nano /data/putnamlab/dbecks/DeNovo_transcriptome/2023_A.pul/functional_annotation/Trinotate/scripts/Trinotate_database_build.sh


#!/bin/bash
#SBATCH --job-name=Trinotate      # Job name
#SBATCH --time=72:00:00                  # Time limit: 3 days
#SBATCH --nodes=1                           # Number of nodes
#SBATCH --ntasks-per-node=1                 # Number of tasks per node
#SBATCH --exclusive                         # Exclusive node allocation
#SBATCH --export=NONE                       # Do not export the environment
#SBATCH --mem=150GB                         # Memory allocation
#SBATCH --mail-type=BEGIN,END,FAIL           # Send email notifications
#SBATCH --mail-user=danielle_becker@uri.edu  # Email address for notifications
#SBATCH --account=putnamlab                 # Account to use
#SBATCH -D /data/putnamlab/dbecks/DeNovo_transcriptome/2023_A.pul/functional_annotation/Trinotate/output
#SBATCH --error="script_error"              # File for error messages
#SBATCH --output="output_script"            # File for standard output

# Load necessary modules
module load Trinotate/4.0.2-foss-2022a
module load DIAMOND/2.1.9-GCC-13.2.0

# Set Trinotate environment variable
export TRINOTATE_HOME=/opt/software/Trinotate/4.0.2-foss-2022a

# Create Trinotate SQLite database and initialize with Diamond
$TRINOTATE_HOME/Trinotate --create \
  --db myTrinotate.sqlite \
  --trinotate_data_dir /data/putnamlab/dbecks/DeNovo_transcriptome/2023_A.pul/functional_annotation/Trinotate/data \
  --use_diamond

sbatch /data/putnamlab/dbecks/DeNovo_transcriptome/2023_A.pul/functional_annotation/Trinotate/scripts/Trinotate_database_build.sh

Submitted batch job 337212
```
