---
layout: post
title: Acropora pulchra Functional Annotation Workflow
date: 2024-01-17
category: [ Code ]
tag: [ RNA-seq, Bioinformatics, Acropora pulchra de novo transcriptome ]
projects: Heatwave - Gametogenesis ; E5 - Molecular Underpinnings
---

## Creating a functional annotation file for *Acropora pulchra* de novo transcriptome.

### **Goal**

The following document contains the bioinformatic pipeline used for the functional annotation of a *de novo* transcriptome for *A. pulchra*. All metadata and information for these projects can be found in this [repository](https://github.com/daniellembecker/A.pul_Heatwave/tree/master) and in these notebook posts of the [*de novo* transcriptome assembly](https://github.com/daniellembecker/DanielleBecker_Lab_Notebook/blob/master/_posts/2023-08-31-Acropora-pulchra-denovo-transcriptome.md) and the [extraction protocol](https://github.com/daniellembecker/DanielleBecker_Lab_Notebook/blob/master/_posts/2023-04-25-Acropora-pulchra-transcriptome-extraction-concentration.md) for the samples used in the assembly. These commands were compiled into bash scripts to run on the URI HPC [Andromeda](https://its.uri.edu/research-computing/using-andromeda/) server.


# Step 1: Obtain *de novo* transcriptome

I am using the *Acropora pulchra* *de novo* transcriptome assembled in [November 2023](https://github.com/daniellembecker/DanielleBecker_Lab_Notebook/blob/master/_posts/2023-08-31-Acropora-pulchra-denovo-transcriptome.md), following methods outlined for [transcript sequence reconstruction from RNA-Seq: reference generation and analysis with Trinity](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3875132/).

Location on Andromeda, the HPC server for URI:
```
cd /data/putnamlab/dbecks/DeNovo_transcriptome/2023_A.pul/data/trinity_out_dir/trinity_out_dir.Trinity.fasta

```

## Make folder structure for functional annotation pipeline

```
cd /data/putnamlab/dbecks/Heatwave_A.pul_2022Project

mkdir functional_annotation
cd functional_annotation

mkdir data
mkdir scripts

```

# Step 2: Open reading frame prediction and protein sequence prediction with Transdecoder and Transpredictor

#### 1) Create .gff3 files for predicted gene structures, .pep for predicted proteins, and .cds predicted coding sequences for your de novo transcriptome

#To include putative gene information in your Trinity analysis, you can use TransDecoder.LongOrfs and TransDecoder.Predict
#First step identifies likely coding regions (long open reading frames or ORFs) in your Trinity transcripts and creates a file named Trinity.fasta.transdecoder_dir/longest_orfs.pep, which contains the predicted protein sequences.

#Second step predicts likely coding regions and identifies potential coding regions using the output from the LongOrfs step and generates several output files in the Trinity.fasta.transdecoder_dir/ directory, including Trinity.fasta.transdecoder.cds (predicted coding sequences) and Trinity.fasta.transdecoder.gff3 (predicted gene structures in GFF3 format).

```
nano /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/functional_annotation/scripts/transdecode_predict.sh

```

```
#!/bin/bash
#SBATCH -t 24:00:00
#SBATCH --nodes=1 --ntasks-per-node=1
#SBATCH --export=NONE
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=danielle_becker@uri.edu
#SBATCH --mem=100GB
#SBATCH --account=putnamlab
#SBATCH -D /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/data/trinity/

module load TransDecoder/5.5.0-foss-2020b

TransDecoder.LongOrfs -t /data/putnamlab/dbecks/DeNovo_transcriptome/2023_A.pul/data/trinity_out_dir/trinity_out_dir.Trinity.fasta

TransDecoder.Predict -t /data/putnamlab/dbecks/DeNovo_transcriptome/2023_A.pul/data/trinity_out_dir/trinity_out_dir.Trinity.fasta

```


```
sbatch /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/functional_annotation/scripts/transdecode_predict.sh

Submitted batch job 292417

#ran well, got .gff3 files, etc. for further gene information

# all files in /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/data/trinity/
```

# Step 3: Download your databases!

Several software programs have the ability to compare sequences to assess homology, the most popular one being NCBI's [BLAST](https://blast.ncbi.nlm.nih.gov/Blast.cgi) (Basic Local Alignment Search Tool - what a great name). BLAST compares nucleotide or protein sequences of interest (called a query) to their sequence databases to find regions of similarity. If a nucleotide or protein sequence of interest significantly matches a sequence/sequences in the databases, BLAST will tag the sequence of interest with the information about the known sequence(s). For example, if a new transcript sequence is identified in a mouse, BLAST could be used to see if any other animals carry a similar sequence, and if they do, what the biological functions of the sequence are.

Other software programs include SWISS-PROT which is a curated protein sequence database that provides a high level of annotation (such as the description of the function of a protein, its domain structure, post-translational modifications, variants, etc), a minimal level of redundancy and a high level of integration with other databases. Recent developments of the database include: an increase in the number and scope of model organisms; cross-references to seven additional databases; a variety of new documentation files; the creation of TREMBL, an unannotated supplement to SWISS-PROT.

#### 1) On the URI HPC Andromeda server, download DIAMOND blast database

In this analysis, the program [DIAMOND BLAST](http://www.diamondsearch.org/index.php) was used. Like regular BLAST, DIAMOND is a sequence aligner for nucleotide and protein sequences; unlike BLAST, it is optimized for a higher performance capability of large datasets at 100x-20,000x speed of BLAST.

On the Andromeda HPC server, download nr database from [NCBI](ftp://ftp.ncbi.nlm.nih.gov/blast/db/FASTA/nr.gz). Convert this database to be Diamond-readable.

This script, downloads the most recent nr database in FASTA format from [NCBI](ftp://ftp.ncbi.nlm.nih.gov/blast/db/FASTA/nr.gz) and uses it to make a Diamond-formatted nr database.
Go to the *sbatch_executables* subdirectory in the Putnam Lab *shared* folder and run the scripts, ```make_diamond_nr_db.sh```  and  ```make_diamond_nr_db.sh``` in the order below.

```
sbatch /data/putnamlab/shared/sbatch_executables/download_nr_database.sh

Submitted batch job 292972

sbatch -d afterok:292972 /data/putnamlab/shared/sbatch_executables/make_diamond_nr_db.sh

Submitted batch job 292973
```

I was getting errors for downloading this step, but I found when I deleted the previous error and output file in the shared folder labeled output.4294967294.out and error.4294967294.out, the scripts run. Will update this output and error syntax so we don't run into the error again. Script ran.

#### 2) On the Andromeda server, download Swiss-Prot database from UniProt/Swiss-Prot.


#### 3) On the Andromeda server, download Trembl database from UniProt/Trembl.


# Step 4: Align query protein sequences against databases

Use prepared translated protein sequences (.pep) and nucleotide predicted coding sequences (.cds) to align with DIAMOND blastp and blastx

In TransDecoder output, the predicted ORFs (Open Reading Frames) and translated protein sequences files are typically named as follows:

- Predicted ORFs:
  - The file containing the predicted ORFs is often named something like your_transcriptome.fasta.transdecoder.pep. Here, your_transcriptome.fasta is the name of your input transcriptome file.


- Translated Protein Sequences:
  - The file containing the translated protein sequences is commonly named something like your_transcriptome.fasta.transdecoder.bed. This file may contain information about the translated sequences, including their coordinates in the original transcriptome.

Pathway to files:

```
/data/putnamlab/dbecks/Heatwave_A.pul_2022Project/data/trinity/trinity_out_dir.Trinity.fasta.transdecoder.pep

/data/putnamlab/dbecks/Heatwave_A.pul_2022Project/data/trinity/trinity_out_dir.Trinity.fasta.transdecoder.bed

```

##### 1) BLAST the protein sequences against NCBI DIAMOND blast

Now that the reference database has been properly generated, the sequences of interest can be aligned against it.

Before aligning, count number of protein sequences:

```
# Count number of protein sequences
zgrep -c ">" /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/data/trinity/trinity_out_dir.Trinity.fasta.transdecoder.pep
523731

#Count number of assembled transcripts
zgrep -c ">" /data/putnamlab/dbecks/DeNovo_transcriptome/2023_A.pul/data/trinity_out_dir/trinity_out_dir.Trinity.fasta
1476390

#Count number of nucleotide coding sequences
zgrep -c ">" /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/data/trinity/trinity_out_dir.Trinity.fasta.transdecoder.cds 523731
```

Write and run script:

```
nano /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/functional_annotation/scripts/apul_diamond_blastp.sh

#!/bin/bash
#SBATCH --job-name="diamond-blastp-protein"
#SBATCH -t 240:00:00
#SBATCH --export=NONE
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=danielle_becker@uri.edu
#SBATCH --mem=100GB
#SBATCH --error="apul_diamond_blastp_out_error"
#SBATCH --output="apul_diamond_blastp_out"
#SBATCH --exclusive
#SBATCH -D /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/functional_annotation/diamond_blast

echo "START" $(date)
module load DIAMOND/2.1.8-GCC-12.3.0 #Load DIAMOND

echo "Updating Acerv annotation" $(date)
diamond blastp -b 2 -d /data/putnamlab/shared/databases/nr.dmnd -q /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/data/trinity/trinity_out_dir.Trinity.fasta.transdecoder.pep -o apul_blastp_annot -f 100 -e 0.00001 -k 2 --threads $SLURM_CPUS_ON_NODE --tmpdir /tmp/

echo "Search complete... converting format to XML and tab"

diamond view -a apul_blastp_annot.daa -o apul_blastp_annot.xml -f 5
diamond view -a apul_blastp_annot.daa -o apul_blastp_annot.tab -f 6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore qlen slen

echo "STOP" $(date)

sbatch /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/functional_annotation/scripts/apul_diamond_blastp.sh

Submitted batch job 293010 20240119

Found that with new version of diamond blast, cannot use diamond view -f 5 option for .xml which is necessary for blast2go downstream, so I made a script with an older version of diamond to use the diamond view command with the -f 5 option to run.

#!/bin/bash
#SBATCH --job-name="diamond-blastp-protein"
#SBATCH -t 240:00:00
#SBATCH --export=NONE
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=danielle_becker@uri.edu
#SBATCH --mem=100GB
#SBATCH --error="error"
#SBATCH --output="out"
#SBATCH --exclusive
#SBATCH -D /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/functional_annotation/diamond_blast

echo "START" $(date)
module load DIAMOND/2.0.7-GCC-10.2.0 #Load DIAMOND

echo "Search complete... converting format to XML and tab"

diamond view -a apul_blastp_annot.daa -o apul_blastp_annot.xml -f 5

echo "STOP" $(date)
```

Output: .tab and .xml file of aligned sequence info. The .xml file is the important one - it will be used as input for Blast2GO

After script has run successfully (took four hours), check how many hits the protein sequences got:

```
wc -l apul_blastp_annot.tab
464867 apul_blastp_annot.tab # 464867 hits out of 523731, 89%
```


Write and run script:

```
nano /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/functional_annotation/scripts/apul_diamond_blastx.sh

#!/bin/bash
#SBATCH --job-name="diamond-blastx-nucleotide"
#SBATCH -t 240:00:00
#SBATCH --export=NONE
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=danielle_becker@uri.edu
#SBATCH --mem=100GB
#SBATCH --error="apul_diamond_blastx_out_error"
#SBATCH --output="apul_diamond_blastx_out"
#SBATCH --exclusive
#SBATCH -D /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/functional_annotation/diamond_blastx

echo "START" $(date)
module load DIAMOND/2.0.7-GCC-10.2.0 #Load DIAMOND

echo "Updating Apul annotation" $(date)

diamond blastx -b 2 -d /data/putnamlab/shared/databases/nr.dmnd -q /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/data/trinity/trinity_out_dir.Trinity.fasta.transdecoder.cds -o apul_blastx_annot -p 8 -k 2 --evalue 1e-5

echo "Search complete... converting format to XML and tab"

diamond view -a apul_blastx_annot.daa -o apul_blastx_annot.xml -f 5
diamond view -a apul_blastx_annot.daa -o apul_blastx_annot.tab -f 6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore qlen slen

echo "STOP" $(date)

sbatch /data/putnamlab/dbecks/Heatwave_A.pul_2022Project/functional_annotation/scripts/apul_diamond_blastx.sh

Submitted batch job 293144 20230122 3:35pm

```

##### i) Secure-copy output files to local computer

```
# From a local terminal window (ie not a remote server)

scp danielle_becker@ssh3.hac.uri.edu:/data/putnamlab/dbecks/Heatwave_A.pul_2022Project/functional_annotation/diamond_blast/apul_blastp_annot.xml /Users/Danielle/Desktop/Putnam_Lab/A.pul_Heatwave/bioinformatics/functional_annotation/diamond

scp danielle_becker@ssh3.hac.uri.edu:/data/putnamlab/dbecks/Heatwave_A.pul_2022Project/functional_annotation/diamond_blast/apul_blastp_annot.tab /Users/Danielle/Desktop/Putnam_Lab/A.pul_Heatwave/bioinformatics/functional_annotation/diamond

scp danielle_becker@ssh3.hac.uri.edu:/data/putnamlab/dbecks/Heatwave_A.pul_2022Project/functional_annotation/diamond_blast/apul_blastx_annot.xml /Users/Danielle/Desktop/Putnam_Lab/A.pul_Heatwave/bioinformatics/functional_annotation/diamond

scp danielle_becker@ssh3.hac.uri.edu:/data/putnamlab/dbecks/Heatwave_A.pul_2022Project/functional_annotation/diamond_blast/apul_blastx_annot.tab /Users/Danielle/Desktop/Putnam_Lab/A.pul_Heatwave/bioinformatics/functional_annotation/diamond
```

DIAMOND BLAST results can now be used in further analyses. To see notes for running Diamond w/ the nr database on all species, go [here](https://github.com/danielle_becker/FunctionalAnnotation/blob/main/scripts/Diamond-NCBI_BLAST.md).
